sdks/go/pkg/beam/runners/dataflow/translate.go

```diff
@@ -454,50 +457,52 @@ func stepID(id int) string {
 	return fmt.Sprintf("s%v", id)
 }
 
-func translateWindow(w *window.Window) proto.Message {
-	// TODO: The only windowing strategy we support is the global window.
-	if w.Kind() != window.GlobalWindow {
-		panic(fmt.Sprintf("Unsupported window type supplied: %v", w))
-	}
-	// We compute the fixed content of this message for use in workflows.
-	msg := rnapi_pb.MessageWithComponents{
+func translateWindowingStrategy(w *window.WindowingStrategy) proto.Message {
+	c := graphx.NewCoderMarshaller()
+	ws := graphx.MarshalWindowingStrategy(c, w)
+
+	msg := &rnapi_pb.MessageWithComponents{
 		Components: &rnapi_pb.Components{
-			Coders: map[string]*rnapi_pb.Coder{
-				"Coder": &rnapi_pb.Coder{
-					Spec: &rnapi_pb.SdkFunctionSpec{
-						Spec: &rnapi_pb.FunctionSpec{
-							Urn: "urn:beam:coders:global_window:0.1",
-						},
-					},
-				},
-			},
+			Coders: c.Build(),
 		},
 		Root: &rnapi_pb.MessageWithComponents_WindowingStrategy{
-			WindowingStrategy: &rnapi_pb.WindowingStrategy{
-				WindowFn: &rnapi_pb.SdkFunctionSpec{
-					Spec: &rnapi_pb.FunctionSpec{
-						Urn: "beam:windowfn:global_windows:v0.1",
-					},
-				},
-				MergeStatus:      rnapi_pb.MergeStatus_NON_MERGING,
-				AccumulationMode: rnapi_pb.AccumulationMode_DISCARDING,
-				WindowCoderId:    "Coder",
-				Trigger: &rnapi_pb.Trigger{
-					Trigger: &rnapi_pb.Trigger_Default_{
-						Default: &rnapi_pb.Trigger_Default{},
-					},
-				},
-				OutputTime:      rnapi_pb.OutputTime_END_OF_WINDOW,
-				ClosingBehavior: rnapi_pb.ClosingBehavior_EMIT_IF_NONEMPTY,
-				AllowedLateness: 0,
-			},
+			WindowingStrategy: ws,
 		},
 	}
-
-	return &msg
+	return msg
 }
 
 func encodeSerializedFn(in proto.Message) (string, error) {
-	// The Beam Runner API uses URL query escaping for serialized fn messages.
-	return protox.EncodeQueryEscaped(in)
+	// The Beam Runner API uses special escaping for serialized fn messages.
+
+	data, err := proto.Marshal(in)
+	if err != nil {
+		return "", err
+	}
+	return encodeString(data), nil
+}
+
+// encodeString is a custom encoding used in some cases by Dataflow.
+//
+// Uses a simple strategy of converting each byte to a single char,
```

Comment:
This is defined as 'percent encoding' or less accurately as 'URL encoding.' 
https://en.wikipedia.org/wiki/Percent-encoding

Pretty sure this is identical to url.PathEscape, perhaps use that?

Reply:
Seems you're right. Good catch.